Wire-iOS/Sources/UserInterface/Components/Controllers/DigitalSignatureVerificationViewController.swift

```diff
@@ -0,0 +1,116 @@
+    }
+    
+    var completion: ((_ result: VoidResult?) -> Void)?
+    
+    private var webView = WKWebView(frame: .zero)
+    private var url: URL?
+    
+    private let success: String = "success"
+    private let failed: String = "failed"
+    
+    init(url: URL) {
+        self.url = url
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupWebView()
+        loadURL()
+    }
+    
+    private func setupWebView() {
+        webView.translatesAutoresizingMaskIntoConstraints = false
+        webView.navigationDelegate = self
+        updateButtonMode()
+        
+        self.view.addSubview(webView)
+        webView.fitInSuperview()
+    }
+    
+    private func updateButtonMode() {
+        let buttonItem = UIBarButtonItem(title: "general.done".localized, style: .done, target: self, action: #selector(DigitalSignatureVerificationViewController.onClose))
+        buttonItem.accessibilityIdentifier = "DoneButton"
+        buttonItem.accessibilityLabel = "general.done".localized
+        buttonItem.tintColor = UIColor.black
+        navigationItem.leftBarButtonItem = buttonItem
+    }
+    
+    private func loadURL() {
+        guard let url = url else { return }
+        let request = URLRequest(url: url)
+        webView.load(request)
+    }
+    
+    @objc private func onClose() {
+        self.dismiss(animated: true, completion: nil)
+    }
+}
+
+extension DigitalSignatureVerificationViewController: WKNavigationDelegate {
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
+        guard 
+            let url = navigationAction.request.url,
+            let response = parse(url) else {
+                decisionHandler(.allow)
+                return
+        }
+        
+        switch response {
+        case .success:
+            completion?(.success)
+            decisionHandler(.cancel)
+        case .failure(let error):
+            completion?(.failure(error))
+            decisionHandler(.cancel)
+        }
+    }
+    
+    func parse(_ url: URL) -> VoidResult? {
+        let urlComponents = URLComponents(string: url.absoluteString)
+        let postCode = urlComponents?.queryItems?.first(where: { $0.name == "postCode" })
+        if let _ = postCode?.value?.range(of: success) {
+            return .success
+        } else if let _ = postCode?.value?.range(of: failed) {
+            guard let error = postCode?.value else {
+                return nil
+            }
+            return error.contains("authentication") ?
+                .failure(VerificationError.authenticationFailed) : .failure(VerificationError.internalServerError)
+        } else {
+            return nil
+        }
```

Comment:
Couldn't we simplify it to something like this?

```suggestion
        let urlComponents = URLComponents(string: url.absoluteString)
        let postCode = urlComponents?.queryItems?.first(where: { $0.name == "postCode" })
        
        switch postCode?.value {
        case "sas-success":
            return .success
        case "sas-error-authentication-failed"
            return .failure(VerificationError.authenticationFailed)
        default:
            return .failure(VerificationError.internalServerError)
        }
```

Reply:
Did like this